src/Controls/src/Core/Xaml/IResourcesLoader.cs

```diff
@@ -8,6 +9,7 @@ namespace Microsoft.Maui.Controls
 {
 	interface IResourcesLoader
 	{
+		[RequiresUnreferencedCode(TrimmerConstants.XamlLoadingTrimmerWarning)]
```

Comment:
Per the discussion in the PR description - if we could remove the interface completely what would the best solution.
Otherwise a good model to follow is:
* Is it expected that the implementation of the method is always going to be trim incompatible? If so, it's OK to annotate the method on the interface.
* Do we have a hard need to be able to call other methods on the interface without RUC anywhere even on implementations which would need RUC on this method? If so, it's basically necessary to add RUC onto the interface method.

In all other cases, I would consider if we can keep the interface without annotations and instead annotate the constructor of the type which needs RUC to implement the interface. Basically: Is the problem the interface itself, or the implementation? Especially if there are multiple implementations which don't need RUC and then there's one which needs it - we should try to avoid annotating the interface.

Reply:
> Is it expected that the implementation of the method is always going to be trim incompatible?

I believe this is the case. It's supposed to take path to a resource and deserialize it into `T`. Theoretically there might be a pre-compiled implementation which maps the paths to `T` without reading the file itself, but as far as I can tell there's just one internal class that implements this internal interface and it is called only from one place.

Would it possibly be cleaner to drop the attribute from the interface method forward the call to a private method with the attribute? Something like:

```c#
public T CreateFromResource<T>(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo) where T : new()
    => CreateFromResourceCore<T>(resourcePath, assembly, lineInfo);

[RequiresUnreferencedCode(TrimmerConstants.XamlLoadingTrimmerWarning)]
private T CreateFromResourceCore<T>(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo) where T : new()
{
    // ...
}
```